### PR TITLE
[MISC] remove includes from version.hpp

### PR DESCRIPTION
**Pull request by SGSSGene**
Change dependency of version.hpp and core/platform.hpp.
This change will inverse the dependency and make core/platform.hpp
include version.hpp.

### CODE REVIEW

File: include/seqan3/core/platform.hpp [hunk 1]
```diff
@@ -70,10 +70,15 @@
...
-#if !__has_include(<seqan3/version.hpp>)
+#if __has_include(<seqan3/version.hpp>)
+#   include <seqan3/version.hpp>
+#else
 #   error SeqAn3 include directory not set correctly. Forgot to add -I ${INSTALLDIR}/include to your CXXFLAGS?
 #endif
 
+
+
+
```
Comment (marehr) [inline include/seqan3/core/platform.hpp:81]:
```suggestion
```

**Comment by marehr** [comment]
Just a note, we need a bit of infrastructure  change to satisfy header test

**Comment by marehr** [comment]
```patch
diff --git a/test/header/CMakeLists.txt b/test/header/CMakeLists.txt
index 89f51d80d..3c0d37e36 100644
--- a/test/header/CMakeLists.txt
+++ b/test/header/CMakeLists.txt
@@ -142,7 +142,8 @@ seqan3_require_benchmark ()
 seqan3_require_test ()
 
 # note: seqan3/std/* will not be tested, because the source files don't have any file extension
-seqan3_header_test (seqan3 "${SEQAN3_CLONE_DIR}/include" "")
+# note: seqan3/version.hpp is one of the only header that is not required to have a seqan3/core/platform.hpp include
+seqan3_header_test (seqan3 "${SEQAN3_CLONE_DIR}/include" "seqan3/version.hpp")
 seqan3_header_test (seqan3_test "${SEQAN3_CLONE_DIR}/test/include" "")
 
 if (SEQAN3_FULL_HEADER_TEST)
```

This should be okay for now :)

**Comment by SGSSGene** [comment]
Very cool. Thank you! Going to try this :-)

**Review by MitraDarja — APPROVED** [review]

**Review by marehr — APPROVED** [review]

**Pull request merged**